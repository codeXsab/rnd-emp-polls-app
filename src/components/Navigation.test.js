import { render } from "@testing-library/react";
import { legacy_createStore as createStore } from "redux";
import { BrowserRouter } from "react-router-dom";
import React from "react";
import { Provider } from "react-redux";
import rootReducer from "../reducers/index";
import middleware from "../middleware/index";
import { setAuthedUser } from "../actions/authedUser";
import Navigation from "./Navigation";
import "@testing-library/jest-dom";


const store = createStore(rootReducer, middleware);

describe("Navbar", () => {
it("should render the component", () => {
    store.dispatch(setAuthedUser({ id:"sarahedo"}));

    const component = render(
    <Provider store={store}>
        <BrowserRouter>
        <Navigation/>
        </BrowserRouter>
    </Provider>
    );
    expect(component).toBeDefined();
    expect(component).toMatchSnapshot();
});

    it("Logout Btn rendered when user is logged in", () => {
    store.dispatch(setAuthedUser({id:"sarahedo"}));

    const component = render(
    <Provider store={store}>
        <BrowserRouter>
        <Navigation/>
        </BrowserRouter>
    </Provider>
    );
    
    const logoutBtn = component.getByTestId("logout-btn");
    expect(logoutBtn).toBeInTheDocument();
    const userSpanElement = component.getByTestId("user-info");
    console.log(userSpanElement);
    expect(userSpanElement.textContent).toBe("");
});
});