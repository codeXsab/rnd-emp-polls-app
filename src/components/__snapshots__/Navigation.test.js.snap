// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar should render the component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav
        class="bg-info p-3 navbar navbar-expand-md navbar-light"
      >
        <a
          class="p-3 navbar-brand"
          href="/"
        >
          <img
            alt="company logo"
            class="d-inline-block align-top"
            height="40"
            src="companyavatar.png"
            width="40"
          />
        </a>
        <button
          aria-controls="responsive-navbar-nav"
          aria-label="Toggle navigation"
          class="navbar-toggler collapsed"
          type="button"
        >
          <span
            class="navbar-toggler-icon"
          />
        </button>
        <div
          class="justify-content-md-between navbar-collapse collapse"
          id="responsive-navbar-nav"
        >
          <div
            class="w-auto navbar-nav"
            data-testid="navbar"
          >
            <li
              class="nav-item"
            >
              <a
                class="m-2 nav-link nav-link active"
                href="/"
              >
                Home
              </a>
            </li>
            <div
              class="nav-item"
            >
              <a
                class="m-2 nav-link"
                href="/leaderboard"
              >
                Leaderboard
              </a>
            </div>
            <div
              class="nav-item"
            >
              <a
                class="m-2 nav-link"
                href="/new"
              >
                New Question
              </a>
            </div>
          </div>
          <div
            class="w-auto navbar-nav"
          >
            <a
              class="w-75 d-flex align-items-center nav-link"
              href="/"
            >
              <div
                class="m-auto d-flex align-items-center container"
              >
                <div>
                  <img
                    alt="Avatar of [object Object]"
                    class="m-2 border border-dark img-fluid rounded-circle"
                    height="40"
                    src=""
                    width="40"
                  />
                </div>
                <div>
                  <span
                    class="ms-3"
                    data-testid="user-info"
                  />
                </div>
              </div>
            </a>
            <a
              class="d-flex align-contents-center fs-6 text-center p-2 nav-link"
              href="/"
            >
              <button
                class="btn-warning btn btn-primary"
                data-testid="logout-btn"
                type="button"
              >
                Sign Out
              </button>
            </a>
          </div>
        </div>
      </nav>
    </div>
  </body>,
  "container": <div>
    <nav
      class="bg-info p-3 navbar navbar-expand-md navbar-light"
    >
      <a
        class="p-3 navbar-brand"
        href="/"
      >
        <img
          alt="company logo"
          class="d-inline-block align-top"
          height="40"
          src="companyavatar.png"
          width="40"
        />
      </a>
      <button
        aria-controls="responsive-navbar-nav"
        aria-label="Toggle navigation"
        class="navbar-toggler collapsed"
        type="button"
      >
        <span
          class="navbar-toggler-icon"
        />
      </button>
      <div
        class="justify-content-md-between navbar-collapse collapse"
        id="responsive-navbar-nav"
      >
        <div
          class="w-auto navbar-nav"
          data-testid="navbar"
        >
          <li
            class="nav-item"
          >
            <a
              class="m-2 nav-link nav-link active"
              href="/"
            >
              Home
            </a>
          </li>
          <div
            class="nav-item"
          >
            <a
              class="m-2 nav-link"
              href="/leaderboard"
            >
              Leaderboard
            </a>
          </div>
          <div
            class="nav-item"
          >
            <a
              class="m-2 nav-link"
              href="/new"
            >
              New Question
            </a>
          </div>
        </div>
        <div
          class="w-auto navbar-nav"
        >
          <a
            class="w-75 d-flex align-items-center nav-link"
            href="/"
          >
            <div
              class="m-auto d-flex align-items-center container"
            >
              <div>
                <img
                  alt="Avatar of [object Object]"
                  class="m-2 border border-dark img-fluid rounded-circle"
                  height="40"
                  src=""
                  width="40"
                />
              </div>
              <div>
                <span
                  class="ms-3"
                  data-testid="user-info"
                />
              </div>
            </div>
          </a>
          <a
            class="d-flex align-contents-center fs-6 text-center p-2 nav-link"
            href="/"
          >
            <button
              class="btn-warning btn btn-primary"
              data-testid="logout-btn"
              type="button"
            >
              Sign Out
            </button>
          </a>
        </div>
      </div>
    </nav>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
